src/port1.0/portbump.tcl

```diff
@@ -280,32 +282,71 @@ proc portbump::bump_main {args} {
                 return -code error "sed sed(1) failed"
             }
 
-            # Check Portfile overwrite permissios
-            if {[catch {file attributes $portfile -permissions u+w} error]} {
-                ui_debug $::errorInfo
-                ui_error "bump: $error"
-                file delete "$tmpfile"
-                return -code error "bump overwrite permissions failed"
-            }
+            if {[tbool ports_bump_patch]} {
+                # Patch mode
+                # Create Portfile.patch
+                # if {[catch {set patchfile [mkstemp "${tempdir}/Portfile.patch"]} error]} {
+                #     ui_debug $::errorInfo
+                #     ui_error "mkstemp: $error"
+                #     return -code error "mkstemp failed"
+                # } else {
+                #     # Extract the Tcl Channel number
+                #     set patchfd [lindex $patchfile 0]
+                #     # Set patchfile to only the file name
+                #     set patchfile [join [lrange $patchfile 1 end]]
+                # }
 
-            # Replace Portfile with Portfile.bump
-            if {[catch {file copy -force $tmpfile $portfile} error]} {
-                ui_debug $::errorInfo
-                ui_error "bump: $error"
-                file delete "$tmpfile"
-                return -code error "bump overwrite failed"
-            }
+                # Set Potfile.patch path
+                set patchfile "${tempdir}/Portfile.patch"
+
+                # Construct diff command
+                set diffcmd {}
+                lappend diffcmd $portutil::autoconf::diff_path -U 0
+                lappend diffcmd $portfile $tmpfile > $patchfile
+
+                # Create and write diff to Portfile.patch
+                if {[catch {exec -ignorestderr -- {*}$diffcmd} error]} {
+                    # Copy Portfile attributes to Portfile.patch
+                    file attributes $portfile {*}$attributes
+
+                    ui_msg "Portfile.patch successfully created at $patchfile"    
+                } else {
+                    ui_error "diff: diff returned 0 exit code (no differences found)"
+                    file delete "$tmpfile"
+                    file delete "$patchfile"
+                    close $tmpfd
+                    close $patchfd
+                    return -code error "diff failed"
```

Comment:
Is that really supposed to be an error? I mean, `port bump` would also just leave the Portfile "unchanged", wouldn't it?

Reply:
Right, now returns 0